src/time.cr

```diff
@@ -605,10 +605,18 @@ struct Time
     {% end %}
 
     unless offset
-      # current TZ may have DST, either in past, present or future
-      ret = LibC.localtime_r(pointerof(second), out tm)
-      raise Errno.new("localtime_r") if ret.null?
-      offset = tm.tm_gmtoff.to_i64
+      {% if !flag?(:windows) %}
+        # current TZ may have DST, either in past, present or future
+        ret = LibC.localtime_r(pointerof(second), out tm)
+        raise Errno.new("localtime_r") if ret.null?
+        offset = tm.tm_gmtoff.to_i64
+      {% else %}
+        # TODO handle daylight?
+        # https://msdn.microsoft.com/en-us/library/windows/desktop/ms724421(v=vs.85).aspx
+        LibWindows.get_time_zone_information(out tz)
```

Comment:
Check the return value here:

    if LibWindows.get_time_zone_information(out tz) == TIME_ZONE_ID_UNKNOWN
      raise WinError.new("GetTimeZoneInformation")
    end

The constants are:

    internal const int TIME_ZONE_ID_INVALID = -1;
    internal const int TIME_ZONE_ID_UNKNOWN = 0;
    internal const int TIME_ZONE_ID_STANDARD = 1;
    internal const int TIME_ZONE_ID_DAYLIGHT = 2;